### DISCUSSION TITLE
[RDKitMinimalLib] Julia

### DISCUSSION THREAD

**Discussion by eloyfelix** [rdkit/rdkit#4644] [Development]
Hi @greglandrum,

RDKit librdkitcffi 2021.09.2 binaries are now available for Julia (via Yggdrasil's binary builder [recipe](https://github.com/JuliaPackaging/Yggdrasil/blob/master/R/RDKit/build_tarballs.jl))

Available both for Linux and OSX (x86_64 and aarch64)

[RDKitMinimalLib.jl](https://github.com/eloyfelix/RDKitMinimalLib.jl) package updated an now also [registered](https://github.com/JuliaRegistries/General/tree/master/R/RDKitMinimalLib) in the Julia General package registry.

I hope it is useful to someone!

**Comment by greglandrum** [comment]
Nice! Thanks for letting us know.

**Reply by eloyfelix** [reply]
RDKit binary builds for Julia are updated to the latest RDKit version (thanks @giordano for your help!) and the Julia package is also updated: https://github.com/eloyfelix/RDKitMinimalLib.jl (including rxn depictions, extra fingeprints...)

It feels it is getting some users in the last month. 
There is a not very convenient way to check the downloads from the Julia registry:

```python
import pandas as pd
import requests
import io

url = "https://julialang-logs.s3.amazonaws.com/public_outputs/current/package_requests.csv.gz"

response = requests.get(url)
df = pd.read_csv(
    io.BytesIO(response.content), sep=",", compression="gzip", index_col=0, quotechar='"',
)

rdml = df[df.index == "44044271-7623-48dc-8250-42433c44e4b7"]
```
<img width="1182" alt="Screenshot 2022-11-02 at 09 38 58" src="https://user-images.githubusercontent.com/1122541/199457607-f2728978-da64-4fb5-be5c-38c4b52f1420.png">

The relevant ones in red (certainly not lots of them).

I also managed to run predictions using the model I wrote a while ago using a feature branch of one of the ONNX libraries. This was not possible a year ago!

```julia
using ONNX
import RDKitMinimalLib: get_mol, get_morgan_fp
import Umlaut: play!
import JSON

path = "chembl_31_multitask.onnx"
targets = JSON.parsefile("targets_31.json")

# dummy input
dummy = rand(Float32, 1024, 1)

# load the model
mt_chembl = ONNX.load(path, dummy)

# load molecule and calc morgan fingerprint
mol = get_mol("CC(=O)Oc1ccccc1C(=O)O")
fp_details = Dict{String, Any}("nBits" => 1024, "radius" => 2)
mfp = get_morgan_fp(mol, fp_details)

# convert the bitstring to a 1024×1 Matrix{Float32}
mfp = map(x->parse(Float32,string(x)),collect(mfp))
mfp = reshape(mfp, (length(mfp), 1))

# test a molecule
out = play!(mt_chembl, mfp)
out = collect(Iterators.flatten(out))

res = tuple.(targets, out)
res = sort(res, by=res->res[2], rev=true)
```

**Comment by eloyfelix** [comment]
MinimalLib binaries for 2022.09.02 already in the Julia registry: https://github.com/JuliaRegistries/General/pull/72838
[Wraps](https://github.com/eloyfelix/RDKitMinimalLib.jl) will by default already install them since it is a patch release.
FYI @greglandrum
